deepchem/data/data_loader.py

```diff
@@ -1753,3 +1755,67 @@ def _featurize_shard(self, shard):
         else:
             x = DFTEntry.create(e_type, true_val, systems)
         return [x]
+
+
+class SAMLoader(DataLoader):
+    """Handles loading of SAM files.
+    Sequence Alignment Map (SAM) is a text-based format used to store 
+    biological sequences aligned to a reference sequence. This
+    class provides convenience files to lead SAM data and
+    extract their features for use in downstream
+    learning tasks.
+    """
+
+    def __init__(self, featurizer: Optional[Featurizer] = None):
+        """Initialize SAMLoader.
+        Parameters
```

Comment:
add new line before the start of parameters, [Numpy style](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) docstrings